### PR TITLE
Reconstruction::MuonReco: more parameters adapted for new yoke geometry

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
- Reconstruction::MuonReco: adapt more parameters to new geometry, as recommended by @bonoxu

ENDRELEASENOTES

### CODE REVIEW

**Comment by webermat** [comment]
ParameterSet checked with mu- at 10 GeV:
The ID efficiency is around 98 % almost everywhere, besides the dip at 90 degrees, which requires more work (90+-0.6 degrees to be precise).
[Mum10_signalIDEff_vs_cosTheta_newMuonSetting170421.pdf](https://github.com/iLCSoft/CLICPerformance/files/951043/Mum10_signalIDEff_vs_cosTheta_newMuonSetting170421.pdf)
The RMS of angular resolution is of the order of 10^-4 rad, relative energy resolution (TruthTracking) of 0.96 %.

**Comment by bonoxu** [comment]
Hi,

I would like to point out that the loss in effiency at ~90 degree is a known fact, due to the gap between left barrel and right barrel in the old Mokka days. I am not sure how the DD software handles the geometry.

Cheers,

**Comment by petricm** [comment]
@bonoxu there is not gap at `z=0` it's one solid volume from `+z` to `-z` (there is no lefr/right barrel)

**Comment by webermat** [comment]
The problem is also not related to tracking or track-cluster matching as well, since a vast majority of the missidentified muons end up as expected as pions (>85 %).

**Comment by bonoxu** [comment]
My next guess (inspired by John) is that the muon yoke is thinner at 90 degree, hence less material to pass through.

**Comment by petricm** [comment]
Sorry but I don't get this comments, at 90 degrees it has the nominal thickness...

**Comment by bonoxu** [comment]
Sorry let me rephrase. The thickness is the same. However, if a particle is passing through at an angle, it experiences more material, comparing to 90 degree.

**Comment by petricm** [comment]
And why would this cause an inefficiency? The path length between between 90 degrees and 91 degrees till the last layer is 1mm on a total thickness of the barrel yoke of 2000mm this is a sub per-mile level.

**Comment by bonoxu** [comment]
Thanks I went back to check the plots again and indeed they can not be explained by the angular arguments.

The issue at exact 90 degree still seems like a detector gap issue to me. In the Mokka simulation of ILD_o1_v5, there are gap regions fed into MarlinPandora. John has made plots showing these regions. These gaps are known to cause a drop in efficiency.

http://www.hep.phy.cam.ac.uk/~xu/ILD_o1_v05_GapsOnly.png
http://www.hep.phy.cam.ac.uk/~xu/ILD_o1_v05_GapsSubDetectors.png

I would suggest to check the implementation of DDMarlinPandora about the gap region. Afterwards, one could also check the existence of tracks, calo hits, and yoke hits. If they all exist, the final test would be to check which cut in one the three Muon pandora algorithms fail. 

I hope this helps.
